images/imagetest/content.go

```diff
@@ -0,0 +1,297 @@
+	m.Children = append(m.Children, layers...)
+	return m
+}
+
+// Blob creates a generic blob with the given data and media type
+// and stores the data in the content store.
+func (tc TestContentStore) Blob(mediaType string, data []byte) TestContent {
+	tc.t.Helper()
+
+	descriptor := ocispec.Descriptor{
+		MediaType: mediaType,
+		Digest:    digest.SHA256.FromBytes(data),
+		Size:      int64(len(data)),
+	}
+
+	ref := string(descriptor.Digest) // TODO: Add random component?
+	if err := content.WriteBlob(tc.ctx, tc.Store, ref, bytes.NewReader(data), descriptor); err != nil {
+		tc.t.Fatal(err)
+	}
+
+	return TestContent{
+		Descriptor: descriptor,
+		Size: Size{
+			Manifest: descriptor.Size,
+			Content:  descriptor.Size,
+		},
+	}
+
+}
+
+// RandomBlob creates a blob object in the content store with random data.
+func (tc TestContentStore) RandomBlob(mediaType string, n int) TestContent {
+	tc.t.Helper()
+
+	data := make([]byte, int64(n))
+	rand.New(rand.NewSource(int64(n))).Read(data)
+
+	return tc.Blob(mediaType, data)
+}
+
+// JSONObject creates an object in the content store by first marshaling
+// to JSON and then storing the data.
+func (tc TestContentStore) JSONObject(mediaType string, i interface{}) TestContent {
+	tc.t.Helper()
+
+	data, err := json.Marshal(i)
+	if err != nil {
+		tc.t.Fatal(err)
+	}
+
+	return tc.Blob(mediaType, data)
+}
+
+// Walk walks all the children of the provided content and calls the provided
+// function with the associated content store.
+// Walk can be used to update an object and reflect that change in the content
+// store.
+func (tc TestContentStore) Walk(c TestContent, fn func(context.Context, *TestContent, content.Store) error) TestContent {
+	tc.t.Helper()
+
+	if err := fn(tc.ctx, &c, tc.Store); err != nil {
+		tc.t.Fatal(err)
+	}
+
+	if len(c.Children) > 0 {
+		children := make([]TestContent, len(c.Children))
+		for i, child := range c.Children {
+			children[i] = tc.Walk(child, fn)
+		}
+		c.Children = children
+	}
+
+	return c
+}
+
+// AddPlatform alters the content desciptor by setting the platform
+func AddPlatform(c TestContent, p ocispec.Platform) TestContent {
+	c.Descriptor.Platform = &p
+	return c
+}
+
+// LimitChildren limits the amount of children in the content.
+// This function is non-recursive and uses the natural ordering.
+func LimitChildren(c TestContent, limit int) TestContent {
+	if images.IsIndexType(c.Descriptor.MediaType) {
+		if len(c.Children) > limit {
+			c.Children = c.Children[0:limit]
+		}
+	}
+	return c
+}
+
+// ContentSize calculates a size property for the test content
+type ContentSize func(TestContent) int64
+
+// SizeOfManifest recursively calculates the manifest reported size
+func SizeOfManifest(t TestContent) int64 {
+	accumulated := t.Size.Manifest
+	for _, child := range t.Children {
+		accumulated = accumulated + SizeOfManifest(child)
+	}
+	return accumulated
+}
+
+// SizeOfContent recursively calculates the size of all stored content
```

Comment:
The function doesn't seem aware of duplicated blobs?

Reply:
Added a comment, since it is just for tests, it would need to be updated for a test which needed to account for them